hat/app/org/hatdex/hat/api/controllers/Authentication.scala

```diff
@@ -223,4 +225,57 @@ class Authentication @Inject() (
         Future.successful(Unauthorized(Json.toJson(ErrorMessage("Invalid Token", "Token does not exist"))))
     }
   }
+
+  /*
+    BEGIN: hat-claim
+    Entire Section below is on HAT Claim.
+    TODO: Refactor to New File?
+   */
+  /**
+   * Sends an email to the owner with a link to claim the hat
+   */
+  def claim(): Action[ApiClaimHatRequest] = UserAwareAction.async(parsers.json[ApiClaimHatRequest]) { implicit request =>
+    val claimHatRequest = request.body
+    val email = request.dynamicEnvironment.ownerEmail
+    val response = Ok(Json.toJson(SuccessResponse("You will shortly receive an email with claim instructions")))
+
+    if (claimHatRequest.email == request.dynamicEnvironment.ownerEmail) {
+      usersService.listUsers.map(_.find(_.roles.contains(Owner()))).flatMap {
+        case Some(user) =>
+          applicationsService.applicationStatus()(request.dynamicEnvironment, user, request).flatMap { applications =>
+            val maybeApplication = applications.find(_.application.id.equals(claimHatRequest.applicationId))
+            val maybeEmailDetails = maybeApplication.map(app => {
+              (app.application.info.name, app.application.info.graphics.logo.normal)
+            })
+
+            // check existing Token  where isSign = true
+
+            if (claimHatRequest.email.equalsIgnoreCase(email)) {
```

Comment:
Looks like a double email check. Isn't it the same check as line 242 above?

Reply:
Fixed.